TempleEight/docs/guide/value-constraints.md

```diff
@@ -0,0 +1,90 @@
+---
+id: value-constraints
+title: Value Constraints
+sidebar_label: Value Constraints
+---
+
+Sometimes you need more control over the values that are acceptable in your data model.
+For these cases Temple provides a suite of Value Constraints: extra metadata you can attach to properties that allow values to be restricted in some way.
+
+## Primitive Restrictions
+
+Many of Temple's primitive data types support optional parameters when you declare them, to give bounds on their values.
+
+An example of this is when an integer is defined. We'll take the `ExampleService` from the [Getting Started](getting-started) guide:
+
+```
+ExampleService: service {
+  foo: string;
+  bar: int;
+}
+```
+
+If, for example, we wanted to constrain the `bar` property to be only positive numbers, we could write:
+
+```
+ExampleService: service {
+  foo: string;
+  bar: int(min: 0);
+}
+```
+
+If we wanted to restrict it between two numbers, for example 0 and 100, we'd write:
+
+```
+ExampleService: service {
+  foo: string;
+  bar: int(100, 0);
+}
+```
+
+Notice that here we didn't provide the argument names to the `int` definitions, as they're optional.
+However, when only a single argument is provided to the `int` definition, it defaults to the `max` argument.
+By using the `min:` name, we can specify the `min` without the `max`.
+
+Most of the primitives in Temple support similar parameters. 
+The full list of parameters can be found in the [primitives](reference/primitives) section of the Templefile reference.
+
+## Annotations
+
+In addition to restricting individual values with primitive parameters, Temple supports adding service-level constraints with annotations.
+
+Currently there is one supported annotation, with more being planned to be released in the future.
+
+### Unique
+
+By adding an `@unique` annotation to any property of a service, then at most one data object stored by the service may have a particular value stored in that property.
```

Comment:
```suggestion
By adding a `@unique` annotation to a property, at most one object stored by the service may have a particular value stored for that property.
```

I believe we use `a` when the start of the word sounds like a consonant, in this case `unique` sounds like `yoonique`. But I don't got an English degree